.github/workflows/notify-merge.yml

```diff
@@ -0,0 +1,20 @@
+name: Notify on Merge
+
+on:
+  push:
+    branches:
+      - stage
+      - main
```

Comment:
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request 

You likely want more something like this
```yaml
on:
  pull_request:
    types:
      - closed

jobs:
  if_merged:
    if: github.event.pull_request.merged == true && branch = stage/main
```